pkg/k8shandler/collection.go

```diff
@@ -0,0 +1,166 @@
+package k8shandler
```

Comment:
do we want to use 'collector' instead of 'collection' to distinguish it from confusion with set/list/collection

Reply:
given the context, i dont think it will be confused with it. It matches with our spec struct name.